src/util.rs

```diff
@@ -0,0 +1,16 @@
+/// Converts a string literal representation of truth to true or false.
+///
+/// Adapted from https://docs.rs/crate/clap_builder/4.5.40/source/src/util/str_to_bool.rs
+pub fn str_to_bool(val: impl AsRef<str>) -> Option<bool> {
+    const TRUE_LITERALS: [&str; 6] = ["y", "yes", "t", "true", "on", "1"];
+    const FALSE_LITERALS: [&str; 6] = ["n", "no", "f", "false", "off", "0"];
+
+    let pat: &str = &val.as_ref().to_lowercase();
```

Comment:
Just side-thinking, wouldn't it be better to use a case-insensitive comparison to search in the array, or perhaps treat is as a `[u8;3]` and put it in that lowercased then look that up?  Very low hanging barely there fruit though, but a string allocation is still an allocation.  Lol.